src/libraries/System.Private.CoreLib/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -2321,7 +2321,11 @@ private static IAlternateEqualityComparer<TAlternateKey, TKey> GetAlternateCompa
              where TAlternateKey : notnull, allows ref struct
         {
             Debug.Assert(IsCompatibleKey<TAlternateKey>(tables));
-            return Unsafe.As<IAlternateEqualityComparer<TAlternateKey, TKey>>(tables._comparer!);
+            // FIXME: review unsafe to confirm correct annotation
+            unsafe
+            {
+                return Unsafe.As<IAlternateEqualityComparer<TAlternateKey, TKey>>(tables._comparer!);
```

Comment:
Please add `unsafe` to `_comparer` and a comment that it must be `IAlternateEqualityComparer<TAlternateKey, TKey>`. Please also remove the unsafe block and add `RequiresUnsafe` to the parent method.